postgres-appliance/postgres_ha.py

```diff
@@ -0,0 +1,286 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import logging
+import re
+import os
+import socket
+import subprocess
+
+import yaml
+import pystache
+import requests
+
+
+def write_certificates(environment):
+    """Write SSL certificate to files
+
+    If certificates are specified, they are written, otherwise
+    dummy certificates are generated and written"""
+
+    ssl_keys = ['SSL_CERTIFICATE', 'SSL_PRIVATE_KEY']
+    if set(ssl_keys) <= set(environment):
+        for k in ssl_keys:
+            with open(environment[k + '_FILE'], 'w') as f:
+                f.write(environment[k])
+    else:
+        openssl_cmd = [
+            '/usr/bin/openssl',
+            'req',
+            '-nodes',
+            '-new',
+            '-x509',
+            '-subj',
+            '/CN=spilo.dummy.org',
+            '-keyout',
+            environment['SSL_PRIVATE_KEY_FILE'],
+            '-out',
+            environment['SSL_CERTIFICATE_FILE'],
+        ]
+        logging.info('Generating ssl certificate')
+        p = subprocess.Popen(openssl_cmd, shell=False, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+        output, _ = p.communicate()
+        logging.debug(output)
+
+    os.chmod(environment['SSL_PRIVATE_KEY_FILE'], 0o600)
+
+
+def deep_update(a, b):
+    """Updates data structures
+
+    Dicts are merged, recursively
+    Lists are appended a+b
+    For anything else, the value of a is returned"""
+
+    if type(a) is dict and type(b) is dict:
+        for key in b:
+            if key in a:
+                a[key] = deep_update(a[key], b[key])
+            else:
+                a[key] = b[key]
+        return a
+    if type(a) is list and type(b) is list:
```

Comment:
I think your list merging might be a little more sophisticated, i.e. to avoid duplicates (note though, that doing it via list(set)) is a bad idea, since in pg_hba case you should care about the order of records. Something like [a + [i in b if i not in a]]